app/views/posts/_posts.html.erb

```diff
@@ -0,0 +1,39 @@
+<% @user_posts.each do |user_post| %>
+    <div class="p-3 mb-3 bg-secondary rounded col-md-8 offset-md-2 text-white">
+      <small>Author: <%=  user_post.user.first_name %> <%=  user_post.user.last_name %></small><br>
+      <%= user_post.post_text %>
+      <% if current_user.id == user_post.user.id %>
```

Comment:
- [ ] There shoud be absolutely NO logic in views.
Rails has built-in mechanism to deal with view-related logic which is View Helpers (app/helpers/*_helper.rb). So the common convention is to move any logic in the views to the helpers. Please take a look at [this](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) for more information.

This can be refactored as follows:
```ruby
def delete_button(post)
  return unless current_user == post.user

  button_to 'delete', post, method: :delete, data: { confirm: 'Are you sure?' }, class: 'btn btn-danger btn-sm'
end
```

Then from your views file you can just call `<%= delete_button(user_post) %>`

Reply:
fixed...moved to post helper